src/os/unix/unix.cpp

```diff
@@ -250,6 +251,9 @@ int CDECL main(int argc, char *argv[])
 	if (argc >= 2 && strncmp(argv[1], "-psn", 4) == 0) {
 		argv[1] = nullptr;
 		argc = 1;
+
+		/* change the directory to the application bundle */
+		chdir(dirname(argv[0]));
```

Comment:
This might break some assumptions about configfile location

Reply:
Yeah, re-reading `launch.sh`, I can see I got it wrong:

```shell
#!/bin/sh		

ROOT_DIR=$(dirname "$0")/..		

export DYLD_LIBRARY_PATH=${ROOT_DIR}/Frameworks		

cd ${ROOT_DIR}/Resources		
exec ./openttd "$@"
```

But the fact that OpenTTD tries to write to its own application bundle is rather surprising, and not guaranteed to work in all situations. But addressing that may be too involved; I'll have to do something else…

Reply:
Turns out this change not only isn't needed, but actually [doesn't even work](https://github.com/macports/macports-ports/blob/c2f29c31270d13afba23ef8eadb506e8770a10ec/_resources/port1.0/group/app-1.0.tcl#L222) on 10.9 and later. So I never actually tested it by launching…

Still, though, OpenTTD appears to detect its location just fine when launched as-is.